[pt2][inductor] retry add `triton.__verison__` as cache key, update cache layout

nmacchioni opened the pull request:
Summary: retry of landing D44550100, try to import triton otherwise consider version as `None`

Test Plan: will make sure windows OSS tests run as well in CI

Differential Revision: D44694213



cc @soumith @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

huydhn reviewed (approved):
Thank you for the fix!  Let me stamp it and let's wait for the test to pass on Windows

nmacchioni commented:
> Thank you for the fix! Let me stamp it and let's wait for the test to pass on Windows

No problem, sorry about that temporary breakage. I added the `ciflow/trunk` so I'll only be able to land after the Windows tests are passing.

nmacchioni committed:
[pt2][inductor] retry add `triton.__verison__` as cache key, update cache layout (#98369)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/98369

retry of landing D44550100, try to import triton otherwise consider version as `None`

Test Plan: will make sure windows OSS tests run as well in CI

Differential Revision: D44694213

fbshipit-source-id: 71083d956b1727fdc9c2d3bf50c3b586bf71aba8

nmacchioni commented:
Landing internally now that the Windows tests are passing
